pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -702,17 +709,27 @@ func conditionallyConvertToIstioMtls(
 	sni string,
 	proxy *model.Proxy,
 	serviceMTLSMode authn_v1alpha1_applier.MutualTLSMode,
+	mtlsReady bool,
 ) *networking.TLSSettings {
 	if tls == nil {
-		if serviceMTLSMode != authn_v1alpha1_applier.MTLSStrict {
-			// Destination service is not in strict-mTLS mode, do nothing.
+		switch serviceMTLSMode {
+		case authn_v1alpha1_applier.MTLSPermissive:
+			// Permissive mTLS enabled and all service instance are labeled mTLSReady
+			if mtlsReady {
```

Comment:
this should be if autoMTLS enabled.

Reply:
Should be both actually, unless the input is already an ANDed of those (but need to change the variable name to be clear)

Reply:
@diemtvu @rshriram poorly named.  I do perform the AND prior to calling this function on line 825